presto-main/src/main/java/com/facebook/presto/sql/planner/ExpressionInterpreter.java

```diff
@@ -220,7 +220,7 @@ public Expression rewriteExpression(Expression node, Void context, ExpressionTre
         analyzer.analyze(rewrite, Scope.create());
 
         // remove syntax sugar
-        rewrite = ExpressionTreeRewriter.rewriteWith(new DesugaringRewriter(analyzer.getExpressionTypes()), rewrite);
+        rewrite = DesugarAtTimeZoneRewriter.rewrite(rewrite, analyzer.getExpressionTypes());
```

Comment:
should be the other desugar rewrite be used here as well?

Reply:
Can't figure it out, we can talk about it outside of the PR. Here, I left the same semantics as before.